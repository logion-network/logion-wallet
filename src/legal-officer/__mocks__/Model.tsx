import {
    fetchRecoveryInfo,
    acceptAccountRecoveryRequest,
    rejectAccountRecoveryRequest,
} from './ModelMock';

export {
    fetchRecoveryInfo,
    acceptAccountRecoveryRequest,
    rejectAccountRecoveryRequest,
};

export function toRecoveryRequestType(value: string | undefined) {
    return value;
}
