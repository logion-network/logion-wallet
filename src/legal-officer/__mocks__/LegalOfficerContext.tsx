import {
    useLegalOfficerContext,
} from "./LegalOfficerContextMock";

export {
    useLegalOfficerContext
}
