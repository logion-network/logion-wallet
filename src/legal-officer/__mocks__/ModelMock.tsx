export let fetchRecoveryInfo = jest.fn();

export function setFetchRecoveryInfo(mockFn: any) {
    fetchRecoveryInfo = mockFn;
}

export let rejectAccountRecoveryRequest = jest.fn();

export function setRejectAccountRecoveryRequest(mockFn: any) {
    rejectAccountRecoveryRequest = mockFn;
}

export let acceptAccountRecoveryRequest = jest.fn();

export function setAcceptAccountRecoveryRequest(mockFn: any) {
    acceptAccountRecoveryRequest = mockFn;
}
