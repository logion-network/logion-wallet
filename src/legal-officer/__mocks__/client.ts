import {
    closeLoc,
    deleteLink,
    voidLoc,
    getVerifiedThirdPartySelections,
} from "./ClientMock";

export {
    closeLoc,
    deleteLink,
    voidLoc,
    getVerifiedThirdPartySelections,
}
