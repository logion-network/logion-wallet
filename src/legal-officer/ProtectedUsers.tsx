import { useLegalOfficerContext } from "./LegalOfficerContext";
import Table, { Cell, EmptyTableMessage } from '../common/Table';
import LegalOfficerName from '../common/LegalOfficerNameCell';
import ProtectedUserDetails from "./ProtectedUserDetails";
import Button from "../common/Button";
import { identityLocDetailsPath } from "./LegalOfficerPaths";
import { useNavigate } from "react-router-dom";

export default function ProtectedUsers() {
    const { activatedProtectionRequests } = useLegalOfficerContext();
    const navigate = useNavigate();

    if(activatedProtectionRequests === null) {
        return null;
    }

    return (
        <Table
            columns={[
                {
                    header: "User",
                    render: request => <Cell content={ `${request.userIdentity.firstName} ${request.userIdentity.lastName}` }/>,
                    align: 'left',
                    width: "55%",
                    renderDetails: request => <ProtectedUserDetails request={ request } />,
                },
                {
                    header: "Other Legal Officer",
                    render: request => <LegalOfficerName address={ request.otherLegalOfficerAddress }/>,
                    width: "45%",
                    align: 'left',
                },
                {
                    header: "Action",
                    render: request => <Button onClick={ () => navigate(identityLocDetailsPath(request.requesterIdentityLoc)) }>Identity LOC</Button>,
                    width: "200px",
                    align: 'center',
                },
            ]}
            data={ activatedProtectionRequests }
            renderEmpty={ () => <EmptyTableMessage>No user to display</EmptyTableMessage> }
        />
    );
}
