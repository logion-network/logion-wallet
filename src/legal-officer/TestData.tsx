import { DateTime } from 'luxon';
import { toIsoString } from "@logion/client";
import { ProtectionRequest } from '@logion/client/dist/RecoveryClient.js';

import { DEFAULT_LEGAL_OFFICER, ANOTHER_LEGAL_OFFICER } from "../common/TestData";

export const PENDING_PROTECTION_REQUESTS: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER.address,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER.address,
        decision: {
            rejectReason: null,
            decisionOn: null,
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(DateTime.fromISO('2021-06-10T11:40:00.000', {zone: "utc"})),
        isRecovery: false,
        addressToRecover: null,
        status: "PENDING"
    }
];

export const PROTECTION_REQUESTS_HISTORY: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER.address,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER.address,
        decision: {
                rejectReason: null,
                decisionOn: toIsoString(DateTime.fromISO('2021-06-10T11:40:00.000', {zone: "utc"})),
                locId: "locId"
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(DateTime.fromISO('2021-06-10T11:40:00.000', {zone: "utc"})),
        isRecovery: false,
        addressToRecover: null,
        status: "ACTIVATED"
    }
];

export const RECOVERY_REQUESTS_HISTORY: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER.address,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER.address,
        decision: {
            rejectReason: null,
            decisionOn: toIsoString(DateTime.fromISO('2021-06-10T11:40:00.000', {zone: "utc"})),
            locId: "locId"
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(DateTime.fromISO('2021-06-10T11:40:00.000', {zone: "utc"})),
        isRecovery: true,
        addressToRecover: "an-address",
        status: "PENDING"
    }
];
