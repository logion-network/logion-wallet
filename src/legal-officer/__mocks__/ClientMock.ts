import { OpenLoc, VerifiedThirdParty } from "@logion/client";

export let closeLoc = jest.fn().mockResolvedValue(undefined);

export function setCloseLocMock(mock: any) {
    closeLoc = mock;
}

export let deleteLink = jest.fn().mockResolvedValue(undefined);

export let voidLoc = jest.fn().mockResolvedValue(undefined);

export function setVoidLocMock(mock: any) {
    voidLoc = mock;
}

let vtpSelections: VerifiedThirdParty[] = [];

export async function getVerifiedThirdPartySelections(params: { locState: OpenLoc } ): Promise<VerifiedThirdParty[]> {
    return Promise.resolve(vtpSelections);
}

export function setVerifiedThirdPartySelections(mock: VerifiedThirdParty[]) {
    vtpSelections = mock;
}

export let requestVote = jest.fn().mockResolvedValue(undefined);

export function setRequestVoteMock(mock: any) {
    requestVote = mock;
}
