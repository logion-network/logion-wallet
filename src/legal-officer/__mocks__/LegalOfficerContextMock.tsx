import { LocType, IdentityLocType } from "@logion/node-api/dist/Types";
import { AxiosInstance } from "axios";

import { COLOR_THEME } from "src/common/TestData";
import { RequestAndLoc } from "../LegalOfficerContext";
import { LegalOfficerData } from "../LegalOfficerData";

export let pendingTokenizationRequests: any[] | null = null;

export let rejectRequest = () => {

};

export let tokenizationRequestsHistory: any[] | null = null;

export let refreshRequests = jest.fn();

export let pendingProtectionRequests: any[] | null = null;

export let activatedProtectionRequests: any[] | null = null;

export let protectionRequestsHistory: any[] | null = null;

export let pendingRecoveryRequests: any[] | null = null;

export let recoveryRequestsHistory: any[] | null = null;

export function setPendingRequests(requests: any[]) {
    pendingTokenizationRequests = requests;
}

export function setRejectRequest(callback: any) {
    rejectRequest = callback;
}

export function setTokenizationRequestsHistory(requests: any[]) {
    tokenizationRequestsHistory = requests;
}

export function setRefreshRequests(callback: any) {
    refreshRequests = callback;
}

export function setPendingProtectionRequests(requests: any[]) {
    pendingProtectionRequests = requests;
}

export function setActivatedProtectionRequests(requests: any[]) {
    activatedProtectionRequests = requests;
}

export function setProtectionRequestsHistory(requests: any[]) {
    protectionRequestsHistory = requests;
}

export function setPendingRecoveryRequests(requests: any[]) {
    pendingRecoveryRequests = requests;
}

export function setRecoveryRequestsHistory(requests: any[]) {
    recoveryRequestsHistory = requests;
}

export let pendingLocRequests: Record<LocType, any[]> | null = null;
export let rejectedLocRequests: Record<LocType, any[]> | null = null;

export let openedLocRequests: Record<LocType, any[]> | null = {
    "Transaction": [],
    "Collection": [],
    "Identity": [],
};

export let closedLocRequests: Record<LocType, any[]> | null = {
    "Transaction": [],
    "Collection": [],
    "Identity": [],
};

export let openedIdentityLocs: RequestAndLoc[] | null = null;

export let openedIdentityLocsByType: Record<IdentityLocType, RequestAndLoc[]> | null = null;

export let closedIdentityLocsByType: Record<IdentityLocType, RequestAndLoc[]> | null = null;

export let voidIdentityLocsByType: Record<IdentityLocType, RequestAndLoc[]> | null = null;

export function setRejectedLocRequests(requests: any[]) {
    rejectedLocRequests = { Collection: [], Transaction: requests, Identity: [] };
}

export function setOpenedLocRequests(requests: any[]) {
    openedLocRequests = { Collection: [], Transaction: requests, Identity: [] };
}

export function setPendingLocRequests(requests: any[]) {
    pendingLocRequests = { Collection: [], Transaction: requests, Identity: [] };
}

export function setClosedLocRequests(requests: any[]) {
    closedLocRequests = { Collection: [], Transaction: requests, Identity: [] };
}

export function setOpenedIdentityLocs(requests: any[]) {
    openedIdentityLocs = requests;
}

export function setOpenedIdentityLocsByType(locs: Record<IdentityLocType, RequestAndLoc[]>) {
    openedIdentityLocsByType = locs;
}

export function setClosedIdentityLocsByType(locs: Record<IdentityLocType, RequestAndLoc[]>) {
    closedIdentityLocsByType = locs;
}

export function setVoidedIdentityLocsByType(locs: Record<IdentityLocType, RequestAndLoc[]>) {
    voidIdentityLocsByType = locs;
}

let onchainSettings: LegalOfficerData = {};

export function setOnchainSettings(settings: LegalOfficerData) {
    onchainSettings = settings;
}

export let refreshLocs = jest.fn();

export function useLegalOfficerContext() {
    return {
        pendingTokenizationRequests,
        rejectRequest,
        tokenizationRequestsHistory,
        refreshRequests,
        pendingProtectionRequests,
        activatedProtectionRequests,
        protectionRequestsHistory,
        colorTheme: COLOR_THEME,
        openedLocRequests,
        openedIdentityLocs,
        openedIdentityLocsByType,
        closedLocRequests,
        closedIdentityLocsByType,
        pendingLocRequests,
        rejectedLocRequests,
        voidIdentityLocsByType,
        axios: {} as AxiosInstance,
        onchainSettings,
        pendingRecoveryRequests,
        refreshLocs,
    };
}
