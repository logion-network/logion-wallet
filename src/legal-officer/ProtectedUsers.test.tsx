jest.mock("../common/CommonContext");
jest.mock('./LegalOfficerContext');

import ProtectedUsers from "./ProtectedUsers";
import { shallowRender } from '../tests';
import { setActivatedProtectionRequests } from "./__mocks__/LegalOfficerContextMock";
import { ACTIVATED_PROTECTION_REQUEST1, ACTIVATED_PROTECTION_REQUEST3 } from "../wallet-user/protection/TestData";

test('renders one row', () => {
    setActivatedProtectionRequests([ ACTIVATED_PROTECTION_REQUEST1 ]);
    const tree = shallowRender(<ProtectedUsers />);
    expect(tree).toMatchSnapshot();
});

test('renders 2 rows', () => {
    setActivatedProtectionRequests([ ACTIVATED_PROTECTION_REQUEST1, ACTIVATED_PROTECTION_REQUEST3 ]);
    const tree = shallowRender(<ProtectedUsers />);
    expect(tree).toMatchSnapshot();
});
