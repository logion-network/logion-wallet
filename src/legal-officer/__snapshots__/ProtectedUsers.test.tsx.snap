// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 2 rows 1`] = `
<Table
  columns={
    Array [
      Object {
        "align": "left",
        "header": "User",
        "render": [Function],
        "renderDetails": [Function],
        "width": "55%",
      },
      Object {
        "align": "left",
        "header": "Other Legal Officer",
        "render": [Function],
        "width": "45%",
      },
      Object {
        "align": "center",
        "header": "Action",
        "render": [Function],
        "width": "200px",
      },
    ]
  }
  data={
    Array [
      Object {
        "addressToRecover": null,
        "createdOn": "2021-06-10T13:48:00.000Z",
        "decision": Object {
          "decisionOn": "2021-06-10T13:48:00.000Z",
          "rejectReason": null,
        },
        "id": "request1",
        "isRecovery": false,
        "legalOfficerAddress": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
        "otherLegalOfficerAddress": "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty",
        "requesterAddress": "5H4MvAsobfZ6bBCDyj5dsrWYLrA8HrRzaqa9p61UXtxMhSCY",
        "requesterIdentityLoc": "86f8884c-79fa-4d80-ba79-b7e3aa4728a7",
        "status": "ACTIVATED",
        "userIdentity": Object {
          "email": "john.doe@logion.network",
          "firstName": "John",
          "lastName": "Doe",
          "phoneNumber": "+1234",
        },
        "userPostalAddress": Object {
          "city": "Liège",
          "country": "Belgium",
          "line1": "Place de le République Française, 10",
          "line2": "boite 15",
          "postalCode": "4000",
        },
      },
      Object {
        "addressToRecover": null,
        "createdOn": "2021-06-10T13:48:00.000Z",
        "decision": Object {
          "decisionOn": "2021-06-10T13:48:00.000Z",
          "rejectReason": null,
        },
        "id": "request3",
        "isRecovery": false,
        "legalOfficerAddress": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
        "otherLegalOfficerAddress": "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty",
        "requesterAddress": "5H4MvAsobfZ6bBCDyj5dsrWYLrA8HrRzaqa9p61UXtxMhSCY",
        "requesterIdentityLoc": "86f8884c-79fa-4d80-ba79-b7e3aa4728a7",
        "status": "ACTIVATED",
        "userIdentity": Object {
          "email": "john.doe@logion.network",
          "firstName": "John",
          "lastName": "Doe",
          "phoneNumber": "+1234",
        },
        "userPostalAddress": Object {
          "city": "Liège",
          "country": "Belgium",
          "line1": "Place de le République Française, 10",
          "line2": "boite 15",
          "postalCode": "4000",
        },
      },
    ]
  }
  renderEmpty={[Function]}
/>
`;

exports[`renders one row 1`] = `
<Table
  columns={
    Array [
      Object {
        "align": "left",
        "header": "User",
        "render": [Function],
        "renderDetails": [Function],
        "width": "55%",
      },
      Object {
        "align": "left",
        "header": "Other Legal Officer",
        "render": [Function],
        "width": "45%",
      },
      Object {
        "align": "center",
        "header": "Action",
        "render": [Function],
        "width": "200px",
      },
    ]
  }
  data={
    Array [
      Object {
        "addressToRecover": null,
        "createdOn": "2021-06-10T13:48:00.000Z",
        "decision": Object {
          "decisionOn": "2021-06-10T13:48:00.000Z",
          "rejectReason": null,
        },
        "id": "request1",
        "isRecovery": false,
        "legalOfficerAddress": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
        "otherLegalOfficerAddress": "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty",
        "requesterAddress": "5H4MvAsobfZ6bBCDyj5dsrWYLrA8HrRzaqa9p61UXtxMhSCY",
        "requesterIdentityLoc": "86f8884c-79fa-4d80-ba79-b7e3aa4728a7",
        "status": "ACTIVATED",
        "userIdentity": Object {
          "email": "john.doe@logion.network",
          "firstName": "John",
          "lastName": "Doe",
          "phoneNumber": "+1234",
        },
        "userPostalAddress": Object {
          "city": "Liège",
          "country": "Belgium",
          "line1": "Place de le République Française, 10",
          "line2": "boite 15",
          "postalCode": "4000",
        },
      },
    ]
  }
  renderEmpty={[Function]}
/>
`;
